### PR TITLE
Update palettizer docs, restructure model examples

### PR DESCRIPTION
**Pull request by crowbat**

### CODE REVIEW

**Review by dengqiaoyu — APPROVED** [review]

File: docs/src/palettization/overview.md [hunk 1]
```diff
@@ -90,4 +90,87 @@ val_metric = validate(prepared_model, val_dataset)
...
 
+Note that `Sensitive K-means` shares the same `KMeansPalettizer` and `KMeansPalettizerConfig` as `Vanilla K-means`. `Sensitive K-means` is achieved simply by passing representative data samples through the model while inside the `calibration_mode` context.
+
 For more details on how to use {class}`~coreai_opt.palettization.config.KMeansPalettizerConfig`, {class}`~coreai_opt.palettization.config.ModuleKMeansPalettizerConfig` to apply different settings to different weights in the model, see [Palettization Config](config.md).
+
+## Training a Palettized model
+
+A `KMeansPalettizer` palettized model can still be fine-tuned in a training pipeline. As palettization is a hard assignment lookup, gradients cannot be propagated for palettized weights, meaning any parameter which is palettized will not update during `optimizer.step()` (the palettization codebook and index assignments will also be fixed).
+
+Any parameters not being palettized can still update and learn so that one can fine-tune any non-palettized portion of the model while being palettization-aware, i.e. incurring palettization noise from any parameters which are palettized.
```
Comment (anotheranshu) [inline docs/src/palettization/overview.md]: "incurring palettization noise" could be construed as a bad thing, when really we expect that doing this will improve accuracy.  Maybe "adapting to parameters which are palettized"?

**Review by anotheranshu — APPROVED** [review]
LGTM!  One minor wording nit
